core/src/test/java/org/apache/calcite/test/MaterializationTest.java

```diff
@@ -588,6 +604,90 @@ private void checkNoMaterialize(String materialize, String query,
     checkMaterialize(mv, query);
   }
 
+  /**
+   * Matching failed because the filtering condition under Aggregate
+   * references columns for aggregation.
+   */
+  @Test public void testAggregate4() {
+    String mv = ""
+        + "select * from\n"
+        + "(select \"deptno\", sum(\"salary\") as \"sum_salary\", sum(\"commission\")\n"
+        + "from " + deduplicatedEmps + "\n"
+        + "group by \"deptno\")\n"
+        + "where \"sum_salary\" > 10";
+    String query = ""
+        + "select * from\n"
+        + "(select \"deptno\", sum(\"salary\") as \"sum_salary\"\n"
+        + "from " + deduplicatedEmps + "\n"
+        + "where \"salary\" > 1000\n"
+        + "group by \"deptno\")\n"
+        + "where \"sum_salary\" > 10";
+    checkNoMaterialize(mv, query, HR_FKUK_MODEL);
+  }
+
+  /**
+   * There will be a compensating Project added after matching of the Aggregate.
+   * This rule targets to test if the Calc can be handled.
+   */
+  @Test public void testCompensatingCalcWithAggregate0() {
+    String mv = ""
+        + "select * from\n"
+        + "(select \"deptno\", sum(\"salary\") as \"sum_salary\", sum(\"commission\")\n"
+        + "from " + deduplicatedEmps + "\n"
+        + "group by \"deptno\")\n"
+        + "where \"sum_salary\" > 10";
+    String query = ""
+        + "select * from\n"
+        + "(select \"deptno\", sum(\"salary\") as \"sum_salary\"\n"
+        + "from " + deduplicatedEmps + "\n"
+        + "group by \"deptno\")\n"
+        + "where \"sum_salary\" > 10";
+    checkMaterialize(mv, query);
+  }
+
+  /**
+   * There will be a compensating Project + Filter added after matching of the Aggregate.
+   * This rule targets to test if the Calc can be handled.
+   */
+  @Test public void testCompensatingCalcWithAggregate1() {
+    String mv = ""
+        + "select * from\n"
+        + "(select \"deptno\", sum(\"salary\") as \"sum_salary\", sum(\"commission\")\n"
+        + "from " + deduplicatedEmps + "\n"
+        + "group by \"deptno\")\n"
+        + "where \"sum_salary\" > 10";
+    String query = ""
+        + "select * from\n"
+        + "(select \"deptno\", sum(\"salary\") as \"sum_salary\"\n"
+        + "from " + deduplicatedEmps + "\n"
+        + "where \"deptno\" >=20\n"
+        + "group by \"deptno\")\n"
+        + "where \"sum_salary\" > 10";
+    checkMaterialize(mv, query);
+  }
+
+  /**
+   * There will be a compensating Project + Filter added after matching of the Aggregate.
```

Comment:
`Project + Filter` means Project on Filter ?

Reply:
Yes, it is.

Reply:
Not sure if it's a convention to use `Project + Filter`, maybe update the doc?